src/pages/enrolment/enrolment.ts

```diff
@@ -84,13 +105,23 @@ export class EnrolmentPage {
       .then(scannedObj => this.authenticate(scannedObj.text))
   }
 
-  submitToken() {
-    this.authenticate(
-      this.authService.getURLFromToken(
-        this.baseURL.trim(),
-        this.tokenName.trim()
+  metaQRHandler() {
+    if (this.baseURL.errors) {
+      this.displayErrorMessage({ statusText: 'Invalid Base URL' })
+      return
+    }
+    if (this.tokenName.errors) {
+      this.displayErrorMessage({ statusText: 'Invalid Token Name' })
+      return
+    }
+    if (!this.baseURL.errors && !this.tokenName.errors) {
```

Comment:
is this `if` clause required ?? The above two are already returning if any errors.

Reply:
Oh let me remove it.